crates/swc_ecma_parser/src/parser/stmt.rs

```diff
@@ -2407,6 +2407,19 @@ export default function waitUntil(callback, options = {}) {
         test_parser(src, Syntax::Es(Default::default()), |p| p.parse_expr());
     }
 
+    #[test]
+    #[should_panic(expected = "Expected ident")]
```

Comment:
No. It should not panic. It's valid JavaScript.

Reply:
@magic-akari i didn't realised 😓  

fixed it, but i'm not sure if the solution accounts for all corner cases...could you take a look?

Reply:
```JavaScript
class Foo {
    static {
        var await = "bar"; // SyntaxError: Unexpected reserved word

        function foo() {
            var await = "bar"; // ok

            class Bar {
                static {
                    var await = "bar"; // SyntaxError: Unexpected reserved word
                }
            }
        }
    }
}
```

I recommend setting the variable `in_static_block` to `false` upon entering the function body. 
Please note that I cannot guarantee that all scenarios have been considered.

Reply:
Fixed, your suggested solution seems much better to me.

Reply:
@magic-akari bump